inc/sitemaps/class-author-sitemap-provider.php

```diff
@@ -35,79 +40,140 @@ public function handles_type( $type ) {
+		if ( empty( $users ) ) {
+			return array();
+		}
 
+		$index      = array();
+		$page       = 1;
+		$user_pages = array_chunk( $users, $max_entries );
+
+		if ( count( $user_pages ) === 1 ) {
+			$page = '';
+		}
+
+		foreach ( $user_pages as $users_page ) {
+
+			$user_id = array_shift( $users_page ); // Time descending, first user on page is most recently updated.
+			$user    = get_user_by( 'id', $user_id );
 			$index[] = array(
-				'loc'     => WPSEO_Sitemaps_Router::get_base_url( 'author-sitemap' . $count . '.xml' ),
-				'lastmod' => '@' . $date, // @ for explicit timestamp format
+				'loc'     => WPSEO_Sitemaps_Router::get_base_url( 'author-sitemap' . $page . '.xml' ),
+				'lastmod' => '@' . $user->_yoast_wpseo_profile_updated, // @ for explicit timestamp format
 			);
+
+			$page++;
 		}
 
 		return $index;
 	}
 
 	/**
+	 * Retrieve users, taking account of all necessary exclusions.
+	 *
+	 * @param array $arguments Arguments to add.
+	 *
+	 * @return array
+	 */
+	protected function get_users( $arguments = array() ) {
+
+		global $wp_version;
+
+		$options = WPSEO_Options::get_all();
+
+		$defaults = array(
+			'who'        => 'authors',
+			'meta_key'   => '_yoast_wpseo_profile_updated',
+			'orderby'    => 'meta_value_num',
+			'order'      => 'DESC',
+			'meta_query' => array(
+				'relation' => 'OR',
+				array(
+					'key'     => 'wpseo_excludeauthorsitemap',
+					'value'   => 'on',
+					'compare' => '!=',
+				),
+				array(
+					'key'     => 'wpseo_excludeauthorsitemap',
+					'compare' => 'NOT EXISTS',
+				),
+			),
+		);
+
+		// TODO Remove version condition after plugin requirements raised to WP 4.3. R.
+		if ( $options['disable_author_noposts'] === true && version_compare( $wp_version, '4.3', '>=' ) ) {
+			$defaults['who']                 = ''; // Otherwise it cancels out next argument.
+			$defaults['has_published_posts'] = true;
+		}
+
+		// TODO Remove version condition after plugin requirements raised to WP 4.4. R.
+		if ( version_compare( $wp_version, '4.4', '>=' ) ) {
+
+			$excluded_roles = $this->get_excluded_roles();
+
+			if ( ! empty( $excluded_roles ) ) {
+				$defaults['who']          = ''; // Otherwise it cancels out next argument.
+				$defaults['role__not_in'] = $excluded_roles;
+			}
+		}
+
+		return get_users( array_merge( $defaults, $arguments ) );
+	}
+
+	/**
+	 * Retrieve array of roles, excluded in settings.
+	 *
+	 * @return array
+	 */
+	protected function get_excluded_roles() {
+
+		static $excluded_roles;
+
+		if ( isset( $excluded_roles ) ) {
+			return $excluded_roles;
+		}
+
+		$options = WPSEO_Options::get_all();
+		$roles   = WPSEO_Utils::get_roles();
+
+		foreach ( $roles as $role_slug => $role_name ) {
+
+			if ( ! empty( $options[ "user_role-{$role_slug}-not_in_sitemap" ] ) ) {
+				$excluded_roles[] = $role_name;
+			}
+		}
+
+		if ( ! empty( $excluded_roles ) ) { // Otherwise it's handled by who=>authors query.
+			$excluded_roles[] = 'Subscriber';
```

Comment:
I don't think it makes a technical difference. Roles are commonly capitalized in text, so just typed that way.